src/viam/sdk/common/exception.cpp

```diff
@@ -0,0 +1,69 @@
+#include <grpc/status.h>
+#include <viam/sdk/common/exception.hpp>
+namespace viam {
+namespace sdk {
+
+ViamException::ViamException(const std::string& what)
+    : std::runtime_error("ViamException: " + what), _error_code(ViamErrorCode::generic){};
+
+ViamException::ViamException(const std::string& type,
+                             const ViamErrorCode& code,
+                             const std::string& what)
+    : std::runtime_error("ViamException(" + type + "): " + what), _error_code(code){};
+
+ViamException::~ViamException() = default;
+ViamException::ViamException(const ::grpc::Status& status_code)
+
+    : ViamException(status_code.error_message() + " " + status_code.error_details()) {}
+
+ViamException ViamException::from_viam_error_code(ViamErrorCode code) {
+    switch (code) {
+        case ViamErrorCode::ok: {
+            return ViamException(
+                "Ran ViamException::from_viam_error_code on an \'ok\' value. Check the code "
+                "value "
```

Comment:
(nit) breaking these up into one word per line is a bit strange looking. Probably good to combine this and the following line.